### PR TITLE
added list for navigating to old survey, delete button for surveys an…

### PR DESCRIPTION
**Pull request by EliasHolmerTietoevry**
In creator you can now list old surveys and navigate to them, you can delete surveys and a button for creating a new survey.

### CODE REVIEW

File: frontend-create-library/src/app/components/list-surveys/list-surveys.component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  constructor(
+    private apiService: ApiControllerService,
+    public dialogRef: MatDialogRef<ListSurveysComponent>
+  ) {}
+
+  ngOnInit() {
+    this.apiService.getSurveys().subscribe((res) => {
+      this.surveys = res;
+      console.log(res);
```
Comment (liviu-florin-ilie) [inline frontend-create-library/src/app/components/list-surveys/list-surveys.component.ts:22]:
```suggestion
```

File: frontend-create-library/src/app/components/survey-creator/survey-creator.component.ts [hunk 2]
```diff
@@ -100,8 +138,9 @@ survey in persistent storage, else populate creator with default starting survey
           console.log('Fetching existing survey');
         },
         error: (err) => {
+          this.creator.text = JSON.stringify(defaultJson);
```
Comment (liviu-florin-ilie) [inline frontend-create-library/src/app/components/survey-creator/survey-creator.component.ts:141]: We should review the way we handle errors. It seems that most of the time the user does not get any feedback on what happened and what can be done in order to overcome the problem.

File: frontend-create-library/src/app/components/survey-creator/survey-creator.component.ts [hunk 3]
```diff
@@ -110,10 +149,94 @@ survey in persistent storage, else populate creator with default starting survey
...
+        title: removeConfirmationTitle,
+        message: removeConfirmationMessage,
+      },
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result === true) {
+        this.apiService.deleteSurvey(Number(this.surveyId)).subscribe({
+          next: (data) => {
+            console.log('Deleting data via autosave:', data);
```
Comment (liviu-florin-ilie) [inline frontend-create-library/src/app/components/survey-creator/survey-creator.component.ts:195]:
```suggestion
     ```

**Review by liviu-florin-ilie — APPROVED** [review]

File: frontend-create-library/src/app/components/survey-creator/survey-creator.component.ts (hunk 2 above)
Comment (EliasHolmerTietoevry) [reply frontend-create-library/src/app/components/survey-creator/survey-creator.component.ts:141]: Yes, should be an overhaul and add some functionality for this.

**Pull request merged**